src/inmanta/server/services/compilerservice.py

```diff
@@ -547,12 +555,16 @@ async def request_recompile(
         remote_id: uuid.UUID,
         metadata: Optional[JsonType] = None,
         env_vars: Optional[Mapping[str, str]] = None,
+        partial: bool = False,
+        removed_resource_sets: Optional[List[str]] = None,
```

Comment:
I am not convinced this is a good idea. We currently don't have a use case for this and I don't think it's a good idea to implement something we don't really need. Once this is part of the API, it will be very hard to remove. @sanderr Could you elaborate on why you think it would be good to have support for this?

Reply:
How will lsm request partial compiles if it's not part of the API? Am I missing something?

Reply:
lsm directly interacts with the compiler service: https://github.com/inmanta/inmanta-lsm/blob/b726fb80038192c9fb80da7aaaf6f72ae6de651c/src/inmanta_lsm/service_inventory.py#L1356. These requests don't go over the API.